lib/src/fake_aggregate_query.dart

```diff
@@ -1,19 +1,128 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:cloud_firestore_platform_interface/cloud_firestore_platform_interface.dart'
+    hide AggregateQuery;
+import 'package:cloud_firestore_platform_interface/cloud_firestore_platform_interface.dart'
+    as platform_interface;
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
 
 import 'fake_aggregate_query_snapshot.dart';
 
 class FakeAggregateQuery implements AggregateQuery {
   final Query _query;
+  final Iterable<AggregateField?> _aggregateFields;
 
-  FakeAggregateQuery(this._query);
+  FakeAggregateQuery(this._query, [this._aggregateFields = const []]);
 
   @override
   Future<AggregateQuerySnapshot> get(
       {AggregateSource source = AggregateSource.server}) async {
     final snapshot = await _query.get();
-    return FakeAggregateQuerySnapshot(_query, snapshot.size);
+    final delegate = _getAggregateQuerySnapshotPlatform(snapshot: snapshot);
+    return FakeAggregateQuerySnapshot(_query, delegate);
   }
 
   @override
   Query<Object?> get query => _query;
+
+  @override
+  AggregateQuery count() => _query.count();
+
+  AggregateQuerySnapshotPlatform _getAggregateQuerySnapshotPlatform({
+    required QuerySnapshot<Object?> snapshot,
+  }) {
+    final dataMaps = snapshot.docs.map((e) => e.data() as Map<String, dynamic>);
```

Comment:
This won't work for converted queries.

In fake_cloud_firestore_test.dart, I added the 'crew' integer field:
```dart
class Movie {
  late String title;
  late int crew;
}

final from =
    (DocumentSnapshot<Map<String, dynamic>> snapshot, _) => snapshot.exists
        ? (Movie()
          ..title = snapshot['title']
          ..crew = snapshot['crew'])
        : null;
final to = (Movie? movie, _) => movie == null
    ? <String, Object?>{}
    : {
        'title': movie.title,
        'crew': movie.crew,
      };
```

Then in mock_query_test.dart, I added this test case:
```dart
test('with Converters', () async {
  final firestore = FakeFirebaseFirestore();
  final movies = firestore
      .collection('movies')
      .withConverter(fromFirestore: from, toFirestore: to);
  await movies.add(Movie()
    ..title = 'Best Movie'
    ..crew = 30);
  await movies.add(Movie()
    ..title = 'Worst Movie'
    ..crew = 20);
  final snapshot =
      await movies.aggregate(sum('crew'), average('crew')).get();
  expect(snapshot.count, 2);
  expect(snapshot.getSum('crew'), 50);
  expect(snapshot.getAverage('crew'), 25);
});
```

I got:
```
type 'Movie' is not a subtype of type 'Map<String, dynamic>' in type cast
package:fake_cloud_firestore/src/fake_aggregate_query.dart 34:56   FakeAggregateQuery._getAggregateQuerySnapshotPlatform.<fn>
dart:_internal                                                     ListIterator.moveNext
package:fake_cloud_firestore/src/fake_aggregate_query.dart 107:27  FakeAggregateQuery.buildAggregateQueryResponseList
package:fake_cloud_firestore/src/fake_aggregate_query.dart 37:12   FakeAggregateQuery._getAggregateQuerySnapshotPlatform
package:fake_cloud_firestore/src/fake_aggregate_query.dart 21:22   FakeAggregateQuery.get
```

I'll merge this because it works well for non-converted queries. We can tackle converted queries in a separate PR.

Reply:
@atn832  I just made another PR to fix this problem, but I am not sure that's a good way to fix it. Please advise.
Thank you very much!😃
https://github.com/atn832/fake_cloud_firestore/pull/291